code/modules/vehicles/wheelchair.dm

```diff
@@ -0,0 +1,107 @@
+/obj/vehicle/ridden/wheelchair //ported from Hippiestation (by Jujumatic)
+	name = "wheelchair"
+	desc = "A chair with big wheels. It looks like you can move in this on your own."
+	icon = 'icons/obj/vehicles.dmi'
+	icon_state = "wheelchair"
+	layer = OBJ_LAYER
+	max_integrity = 100
+	armor = list("melee" = 10, "bullet" = 10, "laser" = 10, "energy" = 0, "bomb" = 10, "bio" = 0, "rad" = 0, "fire" = 20, "acid" = 30)	//Wheelchairs aren't super tough yo
+	legs_required = 0	//You'll probably be using this if you don't have legs
+	canmove = TRUE
+	density = FALSE		//Thought I couldn't fix this one easily, phew
+
+/obj/vehicle/ridden/wheelchair/Initialize()
+	. = ..()
+	var/datum/component/riding/D = LoadComponent(/datum/component/riding)
+	D.vehicle_move_delay = 0
+	D.set_vehicle_dir_layer(SOUTH, OBJ_LAYER)
+	D.set_vehicle_dir_layer(NORTH, ABOVE_MOB_LAYER)
+	D.set_vehicle_dir_layer(EAST, OBJ_LAYER)
+	D.set_vehicle_dir_layer(WEST, OBJ_LAYER)
+
+/obj/vehicle/ridden/wheelchair/ComponentInitialize()	//Since it's technically a chair I want it to have chair properties
+	. = ..()
+	AddComponent(/datum/component/simple_rotation,ROTATION_ALTCLICK | ROTATION_CLOCKWISE, CALLBACK(src, .proc/can_user_rotate),CALLBACK(src, .proc/can_be_rotated),null)
+
+/obj/vehicle/ridden/wheelchair/obj_destruction(damage_flag)
+	new /obj/item/stack/rods(drop_location(), 1)
+	new /obj/item/stack/sheet/metal(drop_location(), 1)
+	if(has_buckled_mobs())
+		var/mob/living/carbon/H = buckled_mobs[1]
+		unbuckle_mob(H)
+	..()
+
+/obj/vehicle/ridden/wheelchair/driver_move(mob/living/user, direction)
+	var/mob/living/carbon/human/H = user
+	if(istype(H))
+		if(!H.get_num_arms() && canmove)
+			to_chat(H, "<span class='warning'>You can't move the wheels without arms!</span>")
+			canmove = FALSE
+			addtimer(CALLBACK(src, .proc/stopmove), 20)
+			return FALSE
+		else
+			var/datum/component/riding/D = LoadComponent(/datum/component/riding)
+			D.vehicle_move_delay = 10/H.get_num_arms()
```

Comment:
1) Should be GetComponent, not LoadComponent since you already added it in Initialize

2) Should be prefering a signal or @ninjanomnom will murder thou

Reply:
I haven't worked with signals before, could you help me implement the second part?